test/dynamo/test_recompiles.py

```diff
@@ -151,3 +151,25 @@ def run_with_automatic():
         with_automatic = run_with_automatic()
         self.assertEqual(with_automatic.frame_count, 3)
         self.assertEqual(with_automatic.op_count, 3)
+
+    def test_aliasing_guard_failures(self):
+        def foo(a, b, c):
+            a.add_(b)
+            return c + 1
+
+        cnt = torch._dynamo.testing.CompileCounter()
+        foo = torch._dynamo.optimize(cnt, nopython=True)(foo)
+
+        x = torch.randn([3])
+        y = torch.randn([3])
+        z = torch.randn([3])
+        foo(x, y, z)
+        self.assertEqual(cnt.frame_count, 1)
+
+        foo(z, y, x)
+        # No recompile, alias preserved
+        self.assertEqual(cnt.frame_count, 1)
+
+        foo(x, y, x)
+        # Recompile, alias changed
+        self.assertEqual(cnt.frame_count, 2)
```

Comment:
Basic question: what's the relationship dynamo's deduping in this PR vs. AOTAutograd's de-duping logic? (Is this PR meant to replace the version of deduping in AOTAutograd?) https://github.com/pytorch/pytorch/blob/999abd56a7be3d73f0d2e9ab06356c7281a3307d/torch/_functorch/aot_autograd.py#L2040

Reply:
This does not cover params - only input tensors guarded through `___check_tensors`

Reply:
Dynamo should be correct standalone, without aotautograd, and AOTAutograd does not guard negative (is not) aliases.

There is a lot of overlap here, and we should probably dedup at some point.

Reply:
@bdhirsh mind if I close this out?

Reply:
yep for sure :) thanks for the context